### PR TITLE
feat: support custom separators for slices

### PR DESCRIPTION
**Pull request by tmzane**

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -7,214 +7,214 @@ import (
 	"time"
 )
 
-type SliceOfBool []bool
+type boolSlice struct {
```
Comment (tmzane) [inline slice.go:10]: Types were renamed for better consistency with exported methods (FlagSet.BoolSlice)

File: slice.go [hunk 2]
```diff
@@ -7,214 +7,214 @@ import (
...
+	sep   string
+	value *[]bool
+}
 
-// Set is flag.Value.Set
-func (sb *SliceOfBool) Set(v string) error {
+// Set implements the flag.Value interface.
+func (s boolSlice) Set(str string) error {
```
Comment (tmzane) [inline slice.go:16]: `v` -> `str` to avoid unnecessary shadowing

File: slice.go [hunk 3]
```diff
@@ -7,214 +7,214 @@ import (
...
-	for i, v := range strings.Split(v, ",") {
-		switch v {
-		case "true", "t", "yes", "y":
-			bools = append(bools, true)
-		case "false", "f", "no", "n":
-			bools = append(bools, false)
-		default:
-			return fmt.Errorf("unknown bool at %d: %s", i, v)
+	for _, v := range strings.Split(str, s.sep) {
+		b, err := strconv.ParseBool(v)
```
Comment (cristaloleg) [inline slice.go:19]: <3

**Review by cristaloleg — APPROVED** [review]
WDYT about panic on empty separator? or make in default to `,` if empty string is provided (I vote for panic :)

**Review by cristaloleg — APPROVED** [review]

**Pull request merged**